### ISSUE TITLE
Consuming a file-per-process file produced on another host fails data check

### ISSUE THREAD

**Issue by CamStan** [LLNL/UnifyFS#675]
### Describe the problem you're observing
Attempting a producer/consumer workflow with the following conditions results in the consumer reading back all zeros instead of the expected data:
- disjoint sets of hosts
- using the UnifyFS `write` followed by `read` example programs
- providing the file-per-process option (`-p nn`)
- providing the date check option (`-k`)

Reproduced this on two nodes and one process per node.

### Describe how to reproduce the problem
Using at least two nodes, start the servers:
```
$UNIFYFS_INSTALL/bin/unifyfs start -S $UNIFYFS_SHAREDFS_DIR -e $UNIFYFS_INSTALL/bin/unifyfsd
```

then run the `write` and `read` examples, excluding the host(s) the other example is run on:
```
echo "Write phase on initial hosts:"
jsrun -n $(($nnodes/2)) -a $nprocs -c18 -r1 -x $latter_hosts -e individual -k $UNIFYFS_LOG_DIR/write-${mode}.err -o $UNIFYFS_LOG_DIR/write-${mode}.out $UNIFYFS_EXAMPLES/write-${mode} -p nn -f write-${mode}.app -v -k

sleep 5

echo "Read phase on latter hosts:"
jsrun -n $(($nnodes/2)) -a $nprocs -c18 -r1 -x $initial_hosts -e individual -k $UNIFYFS_LOG_DIR/read-${mode}.err -o $UNIFYFS_LOG_DIR/read-${mode}.out $UNIFYFS_EXAMPLES/read-${mode} -p nn -f write-${mode}.app -v -k
```

### Include any warning or errors or relevant debugging data

The producer/write is successful.

The consumer/read output is all the data check failures:
```
[0] data check failed at offset 8
[0] read req 0 data check failed
[0] data check failed at offset 1048576
[0] read req 1 data check failed
 ⋮
```

and the consumer/read error log contains zero-filling messages and lipsum check errors:
```
 ⋮
2021-08-27T12:56:06 tid=52234 @ debug_print_read_req() [client_read.c:25] read_req[0x7fffffff5a58] gfid=664735536, file offset=0, length=1048576, buf=0x200028760010 - nread=0, errcode=61 (No data available), byte coverage=[18446744073709551615,18446744073709551615]
2021-08-27T12:56:06 tid=52234 @ process_gfid_reads() [client_read.c:701] zero-filling hole at offset 0 of length 1048576
 ⋮

 ⋮
DEBUG: starting data check
LIPSUM CHECK ERROR: [1] @ offset 8, expected=1 found=0
LIPSUM CHECK ERROR: [0] @ offset 1048576, expected=131072 found=0
 ⋮
```

It appears the data is not getting synced from the producer server to the consumer server.

**Comment by MichaelBrim** [comment]
@CamStan I think there is probably still a bug here, even though skipping the laminate broadcast is intentional. Basically, if a file is not shared, we should still be able to read it by looking up extents at the owner and doing the remote data reads. My guess is there is some logic somewhere that is assuming all laminated files have valid extent metadata at the local server, which is not the case for non-shared files. I'll look to see if I can find the offending code.

**Comment by MichaelBrim** [comment]
@CamStan can you check if the code from PR #676 fixes the issue for you. I'm still trying to get dev working for me on Summit.

**Comment by CamStan** [comment]
@MichaelBrim, your PR appears to have fixed this. 🎉

**Comment by CamStan** [comment]
Fixed by #675
